---
title: "TCGA Genotypes"
output: 
  html_document:
    code_folding: hide
    toc: true
    toc_float: true
    theme: flatly
---


```{r pre-setup, include=FALSE, eval=T}
knitr::opts_chunk$set(fig.width = 5, fig.height = 4, warning=F, message=F, comment=NA, fig.align = 'center')
```

```{r libraries, include=FALSE}
suppressMessages(library(data.table))
suppressMessages(library(dplyr))
suppressMessages(library(ggplot2))
suppressMessages(library(gridExtra))
suppressMessages(library(scales))
theme_set(theme_bw())
```

## Metadata of TCGA Patients

In this notebook, we analyze genotype data from 7000+ TCGA individuals. Much of the processing of genotype data has been done with PLINK v1.9.

Affymetrix SNP6 genotype calls for ~7300 individuals across 20 tumor types were extracted from GDC legacy archive. The 20 tumor types are the ones for which Siming has SSM data for.

Frequency of each tumor type is plotted below. BRCA (breast cancer) is the most frequent, followed by kidney-renal cancer.

```{r tumorTypeFreq}
metadata <- read.delim('../data/TCGA_ALL_metadata.txt', sep='\t', header=F, stringsAsFactors = F, row.names = 1)

tumor.type.freq <- data.frame(table(metadata[,3]))
colnames(tumor.type.freq) <- c('TumorType','Frequency')

tumor.type.freq$TumorType <- factor(tumor.type.freq$TumorType,
                                    levels = tumor.type.freq[order(tumor.type.freq['Frequency']),'TumorType'])


ggplot(tumor.type.freq, aes(x=TumorType, y=Frequency, fill=TumorType)) + 
  geom_bar(stat = 'identity') + 
  xlab('Tumor Type') + ylab('Frequency') + 
  theme(legend.position = 'none', 
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        text = element_text(size=14)) +
  coord_flip() 
```

Demographics of TCGA patients

```{r demo}
demos <- metadata[,6]
demos[demos == 'american indian or alaska native'] <- 'Native Indian'
demos[demos == 'native hawaiian or other pacific islander'] <- 'Pacific Islander'
demos[demos == 'black or african american'] <- 'African American'
demos[demos == 'white'] <- 'European'
demos[demos == 'asian'] <- 'East Asian'
demos[is.na(demos)] <- 'not reported'
metadata[,6] <- demos

demo.freq <- data.frame(table(demos))
colnames(demo.freq) <- c('Population','Frequency')

demo.freq$Population <- factor(demo.freq$Population,
                               levels = demo.freq[order(demo.freq['Frequency']),'Population'])

ggplot(demo.freq, aes(x=Population, y=Frequency, fill=Population)) + 
  geom_bar(stat = 'identity') + 
  xlab('Population') + ylab('Frequency') + 
  theme(legend.position = 'none', 
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        text = element_text(size=16)) +
  scale_fill_manual(values=c('orange','black','blue','red','purple','green')) +
  coord_flip() 
```


## Population Structure

After loading data into PLINK, individuals with less than 95% genotype call-rate were removed to remove low quality samples. Negative strand SNPs were flipped to positive strand.

PCA was performed on 162k LD-pruned SNPs obtained with the `--indep-pairwise 50 5 0.2` flag in PLINK. PC1 seperates European/Asian from African-descent, while PC2 seperates Asian from European/African.

Note that Europeans exhibit population sub-structure. Individuals in the magenta box are retained for downstream analyses.

```{r pca, fig.width=8}
pca.results <- read.delim('../data/plink.eigenvec', 
                          sep = "", 
                          header = F,
                          stringsAsFactors = F) 

pca.results['Population'] <- metadata[pca.results$V2, 6]

ggplot(pca.results, aes(x=V3, y=V4, color=Population)) + 
  geom_point(na.rm = T, alpha=0.4) + 
  xlab('PC 1') + ylab('PC 2') +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        text = element_text(size=16)) + 
  scale_color_manual(values=c('red','blue','green','black','purple','orange')) +
  annotate("rect", xmin=0, xmax=0.007, ymin=-0.006 , ymax=0.005, alpha=0, color='magenta') +
  guides(colour = guide_legend(override.aes = list(alpha = 1)))
```


## Sex-based filtering

The heterozygosity of the X-chromosome was obtained using Plink's `--check-sex`. 389 individuals had discodant sex and were removed.

## SNP-based filtering

SNPs were removed if:

* MAF < 0.05
* Genotype Call-rate < 95%
* HWE test p < 1e-5
* Conflicting strand
* in chrX or chrY

Final dataset with QC retained 4336 individuals, and 631,944 SNPs. 

## Imputation

The Michigan imputation server was used for imputation with the 1000 Genomes Phase 3 reference panel. For each chromosome, PLINK was used to output a VCF file. Eagle was used for phasing and Minimac3 was used for imputation. Minimac3 returns GT (genotypes), DS (dosage), and GP (genotype posterior probabilities) as well as a $R^2$ imputation accuracy. Indels were discarded leaving only SNPs. All imputated SNPs $R^2 < 0.3$ were discarded and the remaining SNPs were annotated with the 1000 Genomes Phase 3 VCF file.

Filtered and annotated VCF files were loaded into plink via `plink2 --vcf chr22.vcf.gz dosage=GP --hard-call-threshold 0.499 --import-dosage-uncertainty 0.5 --id-delim '_' --snps-only --make-bed --out chr22`. The `--hard-threshold-call` sets the maximum distance that the allele dosage can be from the nearest hard-call (too stringent by default). The `--import-dosage-uncertainty` paramter sets the lower-bound on the genotype posterior probability, where uncertain calls are set to missing.

Individual bed files are merged into a single bed file.

I have plotted the genotype posterior probability for chr22 below. 

```{r, genotype_posterior, fig.width=5, fig.height=4}


gp <- read.delim('../data/chr22_GP_field_400k.txt',sep='\t',header=F)

max.gp <- gp %>% mutate(max=pmax(V1, V2, V3))
max.gp.filt <- max.gp[max.gp$max<1,]

ggplot(max.gp, aes(x=max)) + geom_histogram(bins=50, color='black') + 
  xlab('Genotype Posterior Probability') + ylab('Counts') +
  ggtitle('chr22') + 
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        text = element_text(size=16)) +
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
              labels = trans_format("log10", math_format(10^.x))) +
  scale_x_continuous(breaks=seq(0,1,0.1)) + 
  annotation_logticks(sides="l")
```


Further filtering:

* Individuals with genotype call-rate > 95% were kept
* SNPs with genotype call-rate > 95% were kept
* SNPs with MAF > 1% were kept

Final imputed dataset contained **4336** individuals and **~8M** SNPs

A random sub-set of 20k SNPs genome-wide were randomly choosing and their MAF computed and compared with the 1000 Genomes European MAF.

```{r maf, fig.width=5, fig.height=4}

maf_ref <- read.delim('../data/rand20k_1000g_filt.txt',
                      sep='\t',
                      header=F,
                      stringsAsFactors = F,
                      row.names = 1)
maf_ref[maf_ref$V2>0.5,'V2'] = 1 - maf_ref$V2[maf_ref$V2>0.5]

#large file, fread is faster
imputed_maf <- data.frame(fread('../data/imputed_freq.frq',sep=" ", showProgress = F, verbose = F), row.names = 2)

common <- intersect(row.names(maf_ref), row.names(imputed_maf))
maf.df <- data.frame(maf1000G=maf_ref[common,'V2'],
                     mafTCGA=imputed_maf[common, 'MAF'])

ggplot(maf.df, aes(x=maf1000G,y=mafTCGA)) + 
  geom_point(alpha=0.1, color='blue') + 
  xlab('MAF (1000 G)') + ylab('MAF (Imputed TCGA)') + 
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        text = element_text(size=16)) +
  geom_smooth(method='lm',color='red')

```




