<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>TCGA Genotypes</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cancer-Germline</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">TCGA Genotypes</h1>

</div>


<div id="metadata-of-tcga-patients" class="section level2">
<h2>Metadata of TCGA Patients</h2>
<p>In this notebook, we analyze genotype data from 7000+ TCGA individuals. Much of the processing of genotype data has been done with PLINK v1.9.</p>
<p>Affymetrix SNP6 genotype calls for ~7300 individuals across 20 tumor types were extracted from GDC legacy archive. The 20 tumor types are the ones for which Siming has SSM data for.</p>
<p>Frequency of each tumor type is plotted below. BRCA (breast cancer) is the most frequent, followed by kidney-renal cancer.</p>
<pre class="r"><code>metadata &lt;- read.delim(&#39;../data/TCGA_ALL_metadata.txt&#39;, sep=&#39;\t&#39;, header=F, stringsAsFactors = F, row.names = 1)

tumor.type.freq &lt;- data.frame(table(metadata[,3]))
colnames(tumor.type.freq) &lt;- c(&#39;TumorType&#39;,&#39;Frequency&#39;)

tumor.type.freq$TumorType &lt;- factor(tumor.type.freq$TumorType,
                                    levels = tumor.type.freq[order(tumor.type.freq[&#39;Frequency&#39;]),&#39;TumorType&#39;])


ggplot(tumor.type.freq, aes(x=TumorType, y=Frequency, fill=TumorType)) + 
  geom_bar(stat = &#39;identity&#39;) + 
  xlab(&#39;Tumor Type&#39;) + ylab(&#39;Frequency&#39;) + 
  theme(legend.position = &#39;none&#39;, 
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        text = element_text(size=14)) +
  coord_flip() </code></pre>
<p><img src="Figures/tumorTypeFreq-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>Demographics of TCGA patients</p>
<pre class="r"><code>demos &lt;- metadata[,6]
demos[demos == &#39;american indian or alaska native&#39;] &lt;- &#39;Native Indian&#39;
demos[demos == &#39;native hawaiian or other pacific islander&#39;] &lt;- &#39;Pacific Islander&#39;
demos[demos == &#39;black or african american&#39;] &lt;- &#39;African American&#39;
demos[demos == &#39;white&#39;] &lt;- &#39;European&#39;
demos[demos == &#39;asian&#39;] &lt;- &#39;East Asian&#39;
demos[is.na(demos)] &lt;- &#39;not reported&#39;
metadata[,6] &lt;- demos

demo.freq &lt;- data.frame(table(demos))
colnames(demo.freq) &lt;- c(&#39;Population&#39;,&#39;Frequency&#39;)

demo.freq$Population &lt;- factor(demo.freq$Population,
                               levels = demo.freq[order(demo.freq[&#39;Frequency&#39;]),&#39;Population&#39;])

ggplot(demo.freq, aes(x=Population, y=Frequency, fill=Population)) + 
  geom_bar(stat = &#39;identity&#39;) + 
  xlab(&#39;Population&#39;) + ylab(&#39;Frequency&#39;) + 
  theme(legend.position = &#39;none&#39;, 
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        text = element_text(size=16)) +
  scale_fill_manual(values=c(&#39;orange&#39;,&#39;black&#39;,&#39;blue&#39;,&#39;red&#39;,&#39;purple&#39;,&#39;green&#39;)) +
  coord_flip() </code></pre>
<p><img src="Figures/demo-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="population-structure" class="section level2">
<h2>Population Structure</h2>
<p>After loading data into PLINK, individuals with less than 95% genotype call-rate were removed to remove low quality samples. Negative strand SNPs were flipped to positive strand.</p>
<p>PCA was performed on 162k LD-pruned SNPs obtained with the <code>--indep-pairwise 50 5 0.2</code> flag in PLINK. PC1 seperates European/Asian from African-descent, while PC2 seperates Asian from European/African.</p>
<p>Note that Europeans exhibit population sub-structure. Individuals in the magenta box are retained for downstream analyses.</p>
<pre class="r"><code>pca.results &lt;- read.delim(&#39;../data/plink.eigenvec&#39;, 
                          sep = &quot;&quot;, 
                          header = F,
                          stringsAsFactors = F) 

pca.results[&#39;Population&#39;] &lt;- metadata[pca.results$V2, 6]

ggplot(pca.results, aes(x=V3, y=V4, color=Population)) + 
  geom_point(na.rm = T, alpha=0.4) + 
  xlab(&#39;PC 1&#39;) + ylab(&#39;PC 2&#39;) +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        text = element_text(size=16)) + 
  scale_color_manual(values=c(&#39;red&#39;,&#39;blue&#39;,&#39;green&#39;,&#39;black&#39;,&#39;purple&#39;,&#39;orange&#39;)) +
  annotate(&quot;rect&quot;, xmin=0, xmax=0.007, ymin=-0.006 , ymax=0.005, alpha=0, color=&#39;magenta&#39;) +
  guides(colour = guide_legend(override.aes = list(alpha = 1)))</code></pre>
<p><img src="Figures/pca-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="sex-based-filtering" class="section level2">
<h2>Sex-based filtering</h2>
<p>The heterozygosity of the X-chromosome was obtained using Plink’s <code>--check-sex</code>. 389 individuals had discodant sex and were removed.</p>
</div>
<div id="snp-based-filtering" class="section level2">
<h2>SNP-based filtering</h2>
<p>SNPs were removed if:</p>
<ul>
<li>MAF &lt; 0.05</li>
<li>Genotype Call-rate &lt; 95%</li>
<li>HWE test p &lt; 1e-5</li>
<li>Conflicting strand</li>
<li>in chrX or chrY</li>
</ul>
<p>Final dataset with QC retained 4336 individuals, and 631,944 SNPs.</p>
</div>
<div id="imputation" class="section level2">
<h2>Imputation</h2>
<p>The Michigan imputation server was used for imputation with the 1000 Genomes Phase 3 reference panel. For each chromosome, PLINK was used to output a VCF file. Eagle was used for phasing and Minimac3 was used for imputation. Minimac3 returns GT (genotypes), DS (dosage), and GP (genotype posterior probabilities) as well as a <span class="math inline">\(R^2\)</span> imputation accuracy. Indels were discarded leaving only SNPs. All imputated SNPs <span class="math inline">\(R^2 &lt; 0.3\)</span> were discarded and the remaining SNPs were annotated with the 1000 Genomes Phase 3 VCF file.</p>
<p>Filtered and annotated VCF files were loaded into plink via <code>plink --vcf chr22.vcf.gz --vcf-min-gp 0.9 --make-bed --out chr22</code>. The <code>--vcf-min-gp</code> is the minimum genotype posterior probability that a hard-called genotype must fulfill, which was used in <a href="https://www.karger.com/Article/FullText/489758">Shi et al</a> to assess imputation accuracy. Genotypes are set to missing otherwise.</p>
<p>Further filtering:</p>
<ul>
<li>Individuals with genotype call-rate &gt; 95% were kept</li>
<li>SNPs with genotype call-rate &gt; 90% were kept</li>
<li>SNPs with MAF &gt; 5% were kept</li>
</ul>
<p>Final imputed dataset contained <strong>4318</strong> and <strong>~3.2M</strong> SNPs</p>
<p>Validate Allele frequencies with 1000 Genomes MAF.</p>
<pre class="r"><code>maf_ref &lt;- read.delim(&#39;../data/rand20k_1000g_filt.txt&#39;,
                      sep=&#39;\t&#39;,
                      header=F,
                      stringsAsFactors = F,
                      row.names = 1)
maf_ref[maf_ref$V2&gt;0.5,&#39;V2&#39;] = 1 - maf_ref$V2[maf_ref$V2&gt;0.5]

#large file, fread is faster
imputed_maf &lt;- data.frame(fread(&#39;../data/imputed_freq.frq&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

common &lt;- intersect(row.names(maf_ref), row.names(imputed_maf))
maf.df &lt;- data.frame(maf1000G=maf_ref[common,&#39;V2&#39;],
                     mafTCGA=imputed_maf[common, &#39;MAF&#39;])

ggplot(maf.df, aes(x=maf1000G,y=mafTCGA)) + 
  geom_point(alpha=0.1, color=&#39;blue&#39;) + 
  xlab(&#39;MAF (1000 G)&#39;) + ylab(&#39;MAF (Imputed TCGA)&#39;) + 
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        text = element_text(size=16)) +
  geom_smooth(method=&#39;lm&#39;,color=&#39;red&#39;)</code></pre>
<p><img src="Figures/maf-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
<div id="associations" class="section level2">
<h2>Associations</h2>
<div id="null-casecontrol-for-qc" class="section level3">
<h3>Null case/control for QC</h3>
<p>NULL case/control. Samples were partitioned into two random groups for QC purposes. Plink’s <code>--logistic</code> was used to generate p-values.</p>
<pre class="r"><code>random.logis &lt;- data.frame(fread(&#39;../data/Random_Associations.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

p1 &lt;- ggplot(random.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(random.logis$P)

grid.arrange(p1, p2, ncol=2, top=&#39;Random/NULL Partition&#39;)</code></pre>
<p><img src="Figures/nullLogistic-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>Association testing between BRCA vs non-BRCA.</p>
<pre class="r"><code>brca.logis &lt;- data.frame(fread(&#39;../data/BRCA_Associations.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

p1 &lt;- ggplot(brca.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(brca.logis$P)

grid.arrange(p1, p2, ncol=2, top=&#39;BRCA Origin of Tumor&#39;)</code></pre>
<p><img src="Figures/brcaLogistic-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="testing-driver-gene---germline-asssociations" class="section level3">
<h3>Testing driver gene - germline asssociations</h3>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.TP53.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(gene.logis$P)

grid.arrange(p1, p2, ncol=2, top=&#39;TP53&#39;)</code></pre>
<p><img src="Figures/TP53-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.ARID1A.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;ARID1A&#39;)</code></pre>
<p><img src="Figures/ARID1A-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.ARID2.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;ARID2&#39;)</code></pre>
<p><img src="Figures/ARID2-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.ATM.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;ATM&#39;)</code></pre>
<p><img src="Figures/ATM-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.BAP1.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;BAP1&#39;)</code></pre>
<p><img src="Figures/BAP1-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.BRAF.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;BRAF&#39;)</code></pre>
<p><img src="Figures/BRAF-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.CASP8.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;CASP8&#39;)</code></pre>
<p><img src="Figures/CASP8-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.CDH1.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;CDH1&#39;)</code></pre>
<p><img src="Figures/CDH1-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.CDKN2A.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;CDKN2A&#39;)</code></pre>
<p><img src="Figures/CDKN2A-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.CHD4.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;CHD4&#39;)</code></pre>
<p><img src="Figures/CHD4-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.CTCF.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;CTCF&#39;)</code></pre>
<p><img src="Figures/CTCF-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.CTNNB1.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;CTNNB1&#39;)</code></pre>
<p><img src="Figures/CTNNB1-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.EGFR.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;EGFR&#39;)</code></pre>
<p><img src="Figures/EGFR-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.EP300.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;EP300&#39;)</code></pre>
<p><img src="Figures/EP300-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.ERBB2.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;ERBB2&#39;)</code></pre>
<p><img src="Figures/ERBB2-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.ERBB3.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;ERBB3&#39;)</code></pre>
<p><img src="Figures/ERBB3-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.FBXW7.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;FBXW7&#39;)</code></pre>
<p><img src="Figures/FBXW7-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.FGFR2.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;FGFR2&#39;)</code></pre>
<p><img src="Figures/FGFR2-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.IDH1.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;IDH1&#39;)</code></pre>
<p><img src="Figures/IDH1-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.KEAP1.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;KEAP1&#39;)</code></pre>
<p><img src="Figures/KEAP1-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.KRAS.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;KRAS&#39;)</code></pre>
<p><img src="Figures/KRAS-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.MAP3K1.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;MAP3K1&#39;)</code></pre>
<p><img src="Figures/MAP3K1-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.MTOR.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;MTOR&#39;)</code></pre>
<p><img src="Figures/MTOR-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.NCOR1.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;NCOR1&#39;)</code></pre>
<p><img src="Figures/NCOR1-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.NF1.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;NF1&#39;)</code></pre>
<p><img src="Figures/NF1-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.NFE2L2.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;NFE2L2&#39;)</code></pre>
<p><img src="Figures/NFE2L2-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.NOTCH1.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;NOTCH1&#39;)</code></pre>
<p><img src="Figures/NOTCH1-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.NRAS.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;NRAS&#39;)</code></pre>
<p><img src="Figures/NRAS-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.NSD1.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;NSD1&#39;)</code></pre>
<p><img src="Figures/NSD1-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.PBRM1.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;PBRM1&#39;)</code></pre>
<p><img src="Figures/PBRM1-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.PIK3CA.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;PIK3CA&#39;)</code></pre>
<p><img src="Figures/PIK3CA-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.PIK3R1.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;PIK3R1&#39;)</code></pre>
<p><img src="Figures/PIK3R1-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.PPP6C.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;PPP6C&#39;)</code></pre>
<p><img src="Figures/PPP6C-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.PTEN.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;PTEN&#39;)</code></pre>
<p><img src="Figures/PTEN-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.RAC1.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;RAC1&#39;)</code></pre>
<p><img src="Figures/RAC1-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.RB1.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;RB1&#39;)</code></pre>
<p><img src="Figures/RB1-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.SETD2.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;SETD2&#39;)</code></pre>
<p><img src="Figures/SETD2-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.SMAD4.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;SMAD4&#39;)</code></pre>
<p><img src="Figures/SMAD4-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.SPOP.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;SPOP&#39;)</code></pre>
<p><img src="Figures/SPOP-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gene.logis &lt;- data.frame(fread(&#39;../data/germline-somatic/plink.VHL.assoc.logistic.gz&#39;,sep=&quot; &quot;, showProgress = F, verbose = F), row.names = 2)

P &lt;- gene.logis$P
P &lt;- P[!is.na(P)]

p1 &lt;- ggplot(gene.logis, aes(x=P)) + geom_histogram(bins=100, color=&#39;black&#39;, na.rm = T) + xlab(&#39;P-value&#39;) + ylab(&#39;Frequency&#39;)
p2 &lt;- qqunif.plot(P)

grid.arrange(p1, p2, ncol=2, top=&#39;VHL&#39;)</code></pre>
<p><img src="Figures/VHL-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
