<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>TCGA Genotypes</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cancer-Germline</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">TCGA Genotypes</h1>

</div>


<div id="metadata-of-tcga-patients" class="section level2">
<h2>Metadata of TCGA Patients</h2>
<p>In this notebook, we analyze genotype data from 7k+ TCGA individuals. Much of the processing of genotype data has been done with PLINK v1.9.</p>
<p>Affymetrix SNP6 genotype calls for 7000 individuals across 22 tumor types were extracted from GDC legacy archive.</p>
<p>Frequency of each tumor type is plotted below. BRCA (breast cancer) is the most frequent, followed by kidney-renal cancer.</p>
<pre class="r"><code>metadata &lt;- read.delim(&#39;../data/TCGA_ALL_metadata.txt&#39;, 
                       sep=&#39;\t&#39;, 
                       header=F, 
                       stringsAsFactors = F)

tumor.type.freq &lt;- data.frame(table(metadata[,4]))
colnames(tumor.type.freq) &lt;- c(&#39;TumorType&#39;,&#39;Frequency&#39;)

tumor.type.freq$TumorType &lt;- factor(tumor.type.freq$TumorType,
                                    levels = tumor.type.freq[order(tumor.type.freq[&#39;Frequency&#39;]),&#39;TumorType&#39;])


ggplot(tumor.type.freq, aes(x=TumorType, y=Frequency, fill=TumorType)) + 
  geom_bar(stat = &#39;identity&#39;) + 
  xlab(&#39;Tumor Type&#39;) + ylab(&#39;Frequency&#39;) + 
  theme(legend.position = &#39;none&#39;, 
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        text = element_text(size=14)) +
  coord_flip()</code></pre>
<p><img src="TCGA_Analysis_files/figure-html/tumorTypeFreq-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>Demographics of TCGA patients</p>
<pre class="r"><code>demos &lt;- metadata[,7]
demos[demos == &#39;american indian or alaska native&#39;] &lt;- &#39;Native Indian&#39;
demos[demos == &#39;native hawaiian or other pacific islander&#39;] &lt;- &#39;Pacific Islander&#39;
demos[demos == &#39;black or african american&#39;] &lt;- &#39;African American&#39;
demos[demos == &#39;white&#39;] &lt;- &#39;European&#39;
demos[demos == &#39;asian&#39;] &lt;- &#39;East Asian&#39;
demos[is.na(demos)] &lt;- &#39;not reported&#39;
metadata[,7] &lt;- demos

demo.freq &lt;- data.frame(table(demos))
colnames(demo.freq) &lt;- c(&#39;Population&#39;,&#39;Frequency&#39;)

demo.freq$Population &lt;- factor(demo.freq$Population,
                               levels = demo.freq[order(demo.freq[&#39;Frequency&#39;]),&#39;Population&#39;])

ggplot(demo.freq, aes(x=Population, y=Frequency, fill=Population)) + 
  geom_bar(stat = &#39;identity&#39;) + 
  xlab(&#39;Population&#39;) + ylab(&#39;Frequency&#39;) + 
  theme(legend.position = &#39;none&#39;, 
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        text = element_text(size=16)) +
  coord_flip() </code></pre>
<p><img src="TCGA_Analysis_files/figure-html/demo-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="data-qc" class="section level2">
<h2>Data QC</h2>
<p>Data is loaded into PLINK via the following command. PED/MAP files are loaded into PLINK and the individuals sex information is also added. The command <code>--set-hh-missing</code> sets any heterozygous haploid SNPs to be set to missing as they are likely genotyping errors (i.e. male X chromosome SNPs). I also flipped any SNPs on the negative strand.</p>
<pre><code>plink --ped TCGA_ALL.ped
      --map snp6_metadata/snp6_mapfile.map 
      --update-sex ../sex_info.txt 
      --set-hh-missing 
      --missing-genotype N 
      --flip snp6_metadata/NegStrandSNPs.txt
      --make-bed--out TCGA_all</code></pre>
<p>Next, we filter non-European samples, MAF &lt; 5%, SNP call rate &lt; 95%, and individual genotyping rate &lt; 95%. We also perform a sex check between inferred sex and reported sex.</p>
<pre><code>plink --bfile TCGA_all
      --maf 0.05
      --geno 0.05
      --mind 0.05
      --check-sex
      --make-bed --out TCGA_Euro_s1</code></pre>
<p>Next, I remove individuals with inconsistent sex, drop XY chromosomes, and drop SNPs that fail HWE test.</p>
<pre><code>plink --bfile TCGA_Euro_s1 
      --remove sex_problem_samples.txt 
      --exclude snp6_metadata/XY_SnpIDs.txt 
      --hwe 1e-5 
      --make-bed --out TCGA_Euro_QC</code></pre>
</div>
<div id="population-structure" class="section level2">
<h2>Population Structure</h2>
<p>HapMap3 genotype data in PLINK format were downloaded and merged with the TCGA genotype data.</p>
<p>PCA was performed on ~98057 LD-pruned SNPs obtained with the <code>--indep-pairwise 50 2 0.05</code> flag in PLINK.</p>
<p>Individuals that cluster closely with European populations (CEU, YSI) were kept as indicated by the red box.</p>
<pre class="r"><code>pcs &lt;- read.delim(&#39;../data/hapmap_merged_pca.txt&#39;,sep=&#39;&#39;, stringsAsFactors = F, header=F)
meta &lt;- read.delim(&#39;../data/relationships_w_pops_121708.txt&#39;,header=T, stringsAsFactors = F)

pcs$pop &lt;- meta[match(pcs$V2, meta$IID),&#39;population&#39;]
pcs[grepl(&#39;TCGA-&#39;, pcs$V2),&#39;pop&#39;] &lt;- &#39;TCGA Patients&#39;

tcga_df &lt;- pcs[pcs$pop==&#39;TCGA Patients&#39;,c(&#39;V3&#39;, &#39;V4&#39;, &#39;pop&#39;)]
hapmap_df &lt;- pcs[pcs$pop!=&#39;TCGA Patients&#39;,c(&#39;V3&#39;, &#39;V4&#39;, &#39;pop&#39;)]

hapmap_df[&#39;Population&#39;] &lt;- hapmap_df$pop
hapmap_df[hapmap_df$pop==&#39;CEU&#39; | hapmap_df$pop==&#39;TSI&#39;,&#39;Population&#39;] &lt;- &#39;European&#39; 
hapmap_df[hapmap_df$pop==&#39;ASW&#39; | hapmap_df$pop==&#39;YRI&#39; | hapmap_df$pop==&#39;LWK&#39; | hapmap_df$pop==&#39;MKK&#39;,&#39;Population&#39;] &lt;- &#39;African&#39;
hapmap_df[hapmap_df$pop==&#39;GIH&#39; | hapmap_df$pop==&#39;MEX&#39;,&#39;Population&#39;] &lt;- &#39;Indian/Mexican&#39;
hapmap_df[hapmap_df$pop==&#39;CHB&#39; | hapmap_df$pop==&#39;CHD&#39; | hapmap_df$pop==&#39;JPT&#39;,&#39;Population&#39;] &lt;- &#39;Asian&#39;


ggplot() +
  geom_point(data=tcga_df, aes(x=V3, y=V4, color=pop), alpha=0.5) + 
  geom_point(data=hapmap_df, aes(x=V3, y=V4, color=Population), alpha=0.5, size=2) + 
  xlab(&#39;PC 1&#39;) + ylab(&#39;PC 2&#39;) + 
  annotate(&quot;rect&quot;, xmin=-0.005, xmax=0.007, ymin=-0.005, ymax=0.005, alpha=0, color=&#39;red&#39;,size=1) +
  theme(text=element_text(size = 18), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        legend.title = element_blank()) +
  scale_color_manual(values=c(&#39;coral&#39;,&#39;blue&#39;,&#39;green&#39;,&#39;purple&#39;,&#39;black&#39;))</code></pre>
<p><img src="TCGA_Analysis_files/figure-html/pca-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
